packages/playground/src/utils/get_farms.ts

```diff
@@ -1,31 +1,34 @@
-import type { FilterOptions, GridClient, NodeInfo } from "@threefold/grid_client";
+import type { FarmFilterOptions, FarmInfo, GridClient } from "@threefold/grid_client";
+
+import type { Farm } from "@/types";
 
 import { gqlClient, gridProxyClient } from "../clients";
 
 export interface GetFarmsOptions {
   exclusiveFor?: string;
 }
-export async function getFarms(grid: GridClient, filters: FilterOptions, options: GetFarmsOptions = {}) {
-  const nodes: NodeInfo[][] = [];
+export async function getFarms(
+  grid: GridClient,
+  filters: FarmFilterOptions,
+  options: GetFarmsOptions = {},
+): Promise<Farm[]> {
+  const farmsChunks: FarmInfo[][] = [];
 
   let page = 1;
   while (page) {
-    const _nodes = await grid.capacity.filterNodes({ ...filters, page }).catch(() => []);
-    nodes.push(_nodes);
-    page = nodes[page - 1].length ? ++page : 0;
+    const farmsChunk = await grid.capacity.filterFarms({ ...filters, page }).catch(() => []);
+    farmsChunks.push(farmsChunk);
+    page = 0;
   }
```

Comment:
no need for `while` loop if we gonna get only one page. Also in this case we will need to get the rest farms after finishing scrolling the first page